lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -203,6 +193,20 @@ sub create_from_settings {
     return $job;
 }
 
+sub _handle_directly_chained_dep($parent_worker_class, $id, $settings) {
+    if ($parent_worker_class = $parent_worker_class ? $parent_worker_class->value : '') {
+        if (!$settings->{WORKER_CLASS}) {
+            # assume we want to use the worker class from the parent here (and not the default which
+            # is otherwise assumed)
+            $settings->{WORKER_CLASS} = $parent_worker_class;
+        }
+        elsif ($settings->{WORKER_CLASS} ne $parent_worker_class) {
+            die "Specified WORKER_CLASS ($settings->{WORKER_CLASS}) does not match the one from"
+              . " directly chained parent $id ($parent_worker_class)";
+        }
+    }
```

Comment:
```suggestion
    return undef unless $parent_worker_class = $parent_worker_class ? $parent_worker_class->value : '';
    if (!$settings->{WORKER_CLASS}) {
        # assume we want to use the worker class from the parent here (and not the default which
        # is otherwise assumed)
        $settings->{WORKER_CLASS} = $parent_worker_class;
    }
    elsif ($settings->{WORKER_CLASS} ne $parent_worker_class) {
        die "Specified WORKER_CLASS ($settings->{WORKER_CLASS}) does not match the one from"
          . " directly chained parent $id ($parent_worker_class)";
    }
```

though I am not sure if the `… unless $foo = $foo ? … : '';` works syntax-wise.

Reply:
I guess it'll work but maybe it makes sense to do additional changes like this one separately.

Reply:
sure, @b10n1k you can leave that out at your discretion

Reply:
I could try that. But i am afraid that it will need additional tests.

Reply:
More important than syntax wise if if it is readable, and I think it's not.

Reply:
nevermind, leave it out for now.